provider/manifest/manager.go

```diff
@@ -221,6 +237,30 @@ loop:
 
 }
 
+func (m *manager) handleTimeout(req manifestRequest) {
+	x := -1
+	for i, reqEntry := range m.pendingRequests {
+		if reqEntry == req {
+			x = i
+			break
+		}
+	}
+
+	if x != -1 {
+		removed := m.requests[x]
```

Comment:
`removed := m.pendingRequests[x]`?

Reply:
hmm, that is interesting. I need to see why the test is passing now